src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageAnalyzerResultProvider.cs

```diff
@@ -86,24 +89,14 @@ private HashSet<AnalyzeResult> ExtractAnalysisResult(string fileLocation, IDepen
         {
             var results = new HashSet<AnalyzeResult>();
 
-            if (analysisState is null || !analysisState.AreChangesRecommended)
+            if (analysisState is not null && analysisState.AreChangesRecommended)
             {
-                results.Add(new()
-                {
-                    RuleId = Id,
-                    RuleName = this.Name,
-                    FileLocation = fileLocation,
-                    ResultMessage = "No package updates needed",
-                });
```

Comment:
We do not need to emit a Result that is correct.

Reply:
We do need to provide this information to the users that there are no updates needed for them in the current analysis. That is a customer requirement.